### ISSUE TITLE
SSM cacheExpiry

### ISSUE THREAD

**Issue by fdiasr** [middyjs/middy#667]
I am using the middleware, but I am not sure if I need to configure additional parameters to enable cache expiration.

```
ssm({
    fetchData: {
        config: process.env.CONFIG_SSM_PARAMETER,
    },
    cacheExpiry: 30000,
    setToContext: true
})
```

What happens?
--------------
When I made the first request everything is fine, the SSM goes to AWS to fetch data.

The second request, also does a request to AWS.

What were you expecting to happen?
------------------------------------
Second request fetch data from cache

Any other comments?
-------------------
Locally I debug process and the following request I identified the cache variable is empty {}, in each interaction, enforcing middy ssm to access AWS

What versions of software are you using?
----------------------------------------
**Node.js Version:** 12.20.1

**Middy Version:** ^2.1.1

**Comment by willfarrell** [comment]
That is odd, but I was able to create a test that failed on `main`. It seems to be fixed on `feature/cache-error` which includes a refactor to address another edge case. I'm just waiting on some more feedback before merging. Did you want to give that branch a try?

#662

**Comment by fdiasr** [comment]
Ok, I will try it. Also, I give a look at the code and identified the `packages/util/index.js:116`, contains a validation, which I believed is the cause of issue.

https://github.com/middyjs/middy/blob/5da39735af3181ab47328c435b4e8c16600860a7/packages/util/index.js#L116

`if (cached && (cache.expiry >= Date.now() || options.cacheExpiry < 0)) {`

It is necessary validate the `cached.expiry`. Is it correct ?

**Comment by fdiasr** [comment]
For your information, I've tested with the branch `feature/cache-error` and behavior happened as expected. Perfect. I don't know if you need to make more tests to launch a new release or if we can see another PR to fix the current implementation. I am asking it, because I have a project with scheduled release using middy and I d like to keep using it.

**Comment by willfarrell** [comment]
Thanks for doing some extra testing. My goal is to have it merged in and released this weekend.
